cmd/desync/extract_test.go

```diff
@@ -96,3 +100,34 @@ func TestExtractWithFailover(t *testing.T) {
 	_, err = cmd.ExecuteC()
 	require.NoError(t, err)
 }
+
+func TestExtractWithInvalidSeeds(t *testing.T) {
+	outDir, err := ioutil.TempDir("", "")
+	require.NoError(t, err)
+	defer os.RemoveAll(outDir)
+	out := filepath.Join(outDir, "out")
+
+	for _, test := range []struct {
+		name   string
+		args   []string
+		output string
+	}{
+		{"extract with seed directory",
+			[]string{"-s", "testdata/blob1.store", "--seed-dir", "testdata", "testdata/blob1.caibx"}, out},
+		{"extract with corrupted seed",
+			[]string{"--store", "testdata/blob1.store", "--seed", "testdata/blob2_corrupted.caibx", "testdata/blob1.caibx"}, out},
+		{"extract with multiple corrupted seeds",
+			[]string{"--store", "testdata/empty.store", "--seed", "testdata/blob2_corrupted.caibx", "--seed", "testdata/blob1.caibx", "testdata/blob2.caibx"}, out},
+	} {
+		t.Run(test.name, func(t *testing.T) {
+			cmd := newExtractCommand(context.Background())
+			cmd.SetArgs(append(test.args, test.output))
+
+			// Redirect the command's output and run it
+			stderr = ioutil.Discard
+			cmd.SetOutput(ioutil.Discard)
+			_, err := cmd.ExecuteC()
+			require.Error(t, err)
+		})
+	}
```

Comment:
This currently depends on desync detecting any errors in the assembly step. Given the complexities of dealing with corrupted seeds, do you think it'd make sense to validate the output file? Like perhaps read it back from disk and calculating the SHA256 of it, then comparing to the expected value.

Reply:
Sure, something similar to the `validate()` that it's currently in `fileseed.go`, but doing it against the output file.

I guess I can look into it. Would you prefer it to be in this PR or in a followup?

Reply:
In another PR is just fine. You could just hash the whole output file and compare it against a fixed value I suppose

Reply:
The expected hash for the whole file is it already available in the index file or should we calculate that as well?

Reply:
I just noticed that you were commenting on `extract_test`. When you talked about comparing the final hash, were you referring only to the tests cases?

Reply:
Yes, but that can also wait for another PR. I just think given how easy it is to get the seed-algorithms wrong, it'd be good to check that the whole file is assembled correctly in tests. It'd be a little tricky to come up with good test seeds though, since it should cover file-seeds, self-seeds, null-seed etc in specific combinations to exercise all paths properly. On the other hand, having a predicatable seed plan now (it was non-deterministic before) should help.